### PR TITLE
Create symlinks in the upload tarball made by sectionctl deploy

### PR DESCRIPTION
**Pull request by auxesis**
So complex link structures under `node_modules/` work.

### CODE REVIEW

File: commands/deploy.go [hunk 1]
```diff
@@ -244,6 +244,13 @@ func addFileToTarWriter(filePath string, tarWriter *tar.Writer, prefix string) e
 	if err != nil {
 		return err
 	}
+	if stat.Mode()&os.ModeSymlink != 0 {
```
Comment (jstangroome) [inline commands/deploy.go]: Technically more correct where the mask is multibit and good habit so you can ignore how many bits the flag/mask is.
```suggestion
	if stat.Mode()&os.ModeSymlink == os.ModeSymlink {
```

File: commands/deploy.go [hunk 2]
```diff
@@ -254,7 +261,7 @@ func addFileToTarWriter(filePath string, tarWriter *tar.Writer, prefix string) e
 		return fmt.Errorf(fmt.Sprintf("Could not write header for file '%s', got error '%s'", baseFilePath, err.Error()))
 	}
 
-	if !stat.IsDir() {
+	if !stat.IsDir() && stat.Mode()&os.ModeSymlink == 0 {
```
Comment (jstangroome) [inline commands/deploy.go]: Again, per multi-bit comment above.
```suggestion
	if !stat.IsDir() && stat.Mode()&os.ModeSymlink != os.ModeSymlink {
```

**Review by jstangroome — APPROVED** [review]

**Comment by jstangroome** [comment]
At what point of having reimplemented much of tar in our Go code do we instead insist that the user has `tar` on their PATH and we then shell out? :)

**Comment by auxesis** [comment]
To be honest, given the scope of what we're trying to do here with compressed archives, shelling out to `tar` might actually be more maintainable. 

One potential problem I can see is the differences between GNU and BSD tar – we'd need to use a subset of features and syntax that both support.

**Comment by jstangroome** [comment]
> One potential problem I can see is the differences between GNU and BSD tar – we'd need to use a subset of features and syntax that both support.

Looks like Windows has the BSD flavour of tar:
https://techcommunity.microsoft.com/t5/containers/tar-and-curl-come-to-windows/ba-p/382409

**Pull request merged**